pkg/common/resource_utils/service.go

```diff
@@ -108,13 +125,20 @@ func BuildExternalService(src *srapi.StarRocksCluster, name string, serviceType
 
 	var ports []corev1.ServicePort
 	for _, sp := range srPorts {
-		ports = append(ports, corev1.ServicePort{
+		servicePort := corev1.ServicePort{
 			Name:       sp.Name,
 			Port:       sp.Port,
 			NodePort:   sp.NodePort,
 			Protocol:   corev1.ProtocolTCP,
 			TargetPort: intstr.FromInt(int(sp.ContainerPort)),
-		})
+		}
+		if servicePort.Name == FeQueryPortName {
```

Comment:
how about the webPort? does it need to add appProtocol `http`?

Reply:
didn't see our webport related to the server-first-protocol document. HTTP protocol doesn't look like a server-first-protocol per the description in the doc.

Reply:
MySQL is a server-first-protocol, so we only need to add appProtocol to the query port, 9030.